src/lib/__tests__/InstantSearch-test.js

```diff
@@ -1576,6 +1577,62 @@ describe('use', () => {
   });
 });
 
+describe('unuse', () => {
+  it('unsubscribes middleware', async () => {
+    const indexName = 'indexName';
+    const searchClient = createSearchClient();
+    const search = new InstantSearch({
+      indexName,
+      searchClient,
+    });
+    const middlewareSpy1 = {
+      onStateChange: jest.fn(),
+      subscribe: jest.fn(),
+      unsubscribe: jest.fn(),
+    };
+    const middleware1 = jest.fn(() => middlewareSpy1);
+    const middlewareSpy2 = {
+      onStateChange: jest.fn(),
+      subscribe: jest.fn(),
+      unsubscribe: jest.fn(),
+    };
+    const middleware2 = jest.fn(() => middlewareSpy2);
+
+    search.addWidgets([
+      searchBoxOriginal({ container: document.createElement('div') }),
```

Comment:
```suggestion
      connectSearchBox(() => {})(),
```

this is consistent with the other tests